### ISSUE TITLE
TST/MAINT:  new test failures on latest py 2.7 version

### ISSUE THREAD

**Issue by josef-pkt** [statsmodels/statsmodels#3228]
a series of anova tests fail because of an `I/O operation on closed file`
this showed up recently in PR test runs, it only affects the python 2.7 with latest versions
somebody is now closing files or BytesIO, but I haven't checked

https://travis-ci.org/statsmodels/statsmodels/jobs/165800260

```
======================================================================
ERROR: test suite for <class 'statsmodels.stats.tests.test_anova.TestAnova3HC0'>
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda2/envs/statsmodels-test/lib/python2.7/site-packages/nose/suite.py", line 209, in run
    self.setUp()
  File "/home/travis/miniconda2/envs/statsmodels-test/lib/python2.7/site-packages/nose/suite.py", line 292, in setUp
    self.setupContext(ancestor)
  File "/home/travis/miniconda2/envs/statsmodels-test/lib/python2.7/site-packages/nose/suite.py", line 315, in setupContext
    try_run(context, names)
  File "/home/travis/miniconda2/envs/statsmodels-test/lib/python2.7/site-packages/nose/util.py", line 471, in try_run
    return func()
  File "/home/travis/miniconda2/envs/statsmodels-test/lib/python2.7/site-packages/statsmodels-0.8.0-py2.7-linux-x86_64.egg/statsmodels/stats/tests/test_anova.py", line 79, in setupClass
    kidney_table.seek(0)
  File "/home/travis/miniconda2/envs/statsmodels-test/lib/python2.7/StringIO.py", line 104, in seek
    _complain_ifclosed(self.closed)
  File "/home/travis/miniconda2/envs/statsmodels-test/lib/python2.7/StringIO.py", line 40, in _complain_ifclosed
    raise ValueError, "I/O operation on closed file"
ValueError: I/O operation on closed file
```

**Comment by yarikoptic** [comment]
the same problem showed up in debian now... https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=841610
For some reason I have a feeling that I saw such problem before but can't recall..

**Comment by josef-pkt** [comment]
This bug is still in debian testing, fixed in pandas 0.19.1, #3239 dropped it after pandas was fixed

I'm looking at stats test_anova.py. I don't see a reason why we have to parse the StringIO several times. We can create a module global kidney_table DataFrame instead.

The other failures are in base/test_generic_methods.py but those might all be imported from stats.test_anova. I was reusing the test classes from there in the unit tests.
So, it looks like there is nothing that has to be fixed there.

```
    def setup_class(cls):
        import statsmodels.stats.tests.test_anova as ttmod

        test = ttmod.TestAnova3()
        test.setupClass()
```
